prow/cmd/audit-owners/main.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2018 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This application reads kubernetes OWNERS files within a repo and compares it
+// to collaborators for a repo to find those listed in OWNERS files who are not
+// collaborators
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+	"k8s.io/test-infra/prow/github"
+)
+
+// The root location of the repo to start walking the filesystem
+var repoRoot string
+
+// The name of the GitHub org to use when querying for collaborators
+var githubOrg string
+
+// The name of the GitHub repo to use when querying for collaborators
+var githubRepo string
+
+func init() {
+	flag.StringVar(&repoRoot, "s", "", "The location of the repo on the local filesystem to start inspecting")
+	flag.StringVar(&githubOrg, "o", "", "The name of the GitHub org to use when querying for collaborators")
+	flag.StringVar(&githubRepo, "r", "", "The name of the GitHub repo to use when querying for collaborators")
+}
+
+func main() {
+
+	flag.Parse()
+
+	// We require a github token to make API calls due to rate limiting
+	if os.Getenv("GITHUB_TOKEN") == "" {
```

Comment:
For consistency with other test-infra tools this might be better configured via a flag that specifies a file containing the token.